File: gpfksdlrn/level1/추억_점수.js
// https://school.programmers.co.kr/learn/courses/30/lessons/176963

function solution(name, yearning, photo) {
  const score = new Map();

  name.forEach((person, i) => {
    score.set(person, yearning[i]);
  });

  return photo.map((people) => {
    let sum = 0;

    for (const person of people) {
      sum += score.get(person) ?? 0;
    }

    return sum;
  });
}

File: gnsals0904/level1/추억_점수.java
import java.util.*;

class Solution {
    static Map<String, Integer> score = new HashMap<>();
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        for (int i = 0; i < name.length; i++) {
            score.put(name[i], yearning[i]);
        }
        
        int[] answer = new int[photo.length];
        
        for (int i = 0; i < photo.length; i++) {
            int s = 0;

            for (String person : photo[i]) {
                if (score.containsKey(person)) {
                    s += score.get(person);
                }
            }
            answer[i] = s;
        }
        return answer;
    }
}
